plugins/@grouparoo/pipedrive/src/lib/common/destinationMappingOptions.ts

```diff
@@ -1,62 +1,79 @@
-      true
-    );
+export type KnownPipedriveField = {
+  key: string;
+  pipedriveKey: string;
+  type: DestinationMappingOptionsResponseType;
+  required: boolean;
+  important: boolean;
+  groupMembershipField: boolean;
+};
 
-    const required = allKnown.filter((field) => field.required);
+export const fetchKnownFields = async (
+  type: EntityType,
+  client: PipedriveClient
+): Promise<KnownPipedriveField[]> => {
+  const required = configMapping[type].requiredFields;
+  const allowedFields = configMapping[type].allowedFields;
 
-    // Filter out required fields and those that we create to manage group membership
-    const knownNotRequired = allKnown.filter(
-      (field) => !field.required && !field.groupMembershipField
-    );
+  // Allow setting "End of" for date ranges and "Currency" for monetary fields
+  // The keys for these are like cd77a9b30948242ef1ef57acbe28c0aa054b8fde_until
+  const subfieldsRegex = /(_until|_currency)$/;
 
-    return {
-      labels: {
-        property: {
-          singular: "Pipedrive Field",
-          plural: "Pipedrive Fields",
-        },
-        group: {
-          singular: "Pipedrive Filter",
-          plural: "Pipedrive Filters",
-        },
-      },
-      properties: {
-        known: knownNotRequired,
-        required,
-        allowOptionalFromProperties: false,
-      },
-    };
-  };
+  const out: KnownPipedriveField[] = [];
+
+  const fields = await configMapping[type]
+    .getEntityCalls(client)
+    .fields.getAll();
+
+  for (const field of fields) {
+    // Some fields are system fields that don't make sense to manually update (counts, computed fields, auto timestamps...)
+    // There's no explicit flag for this, but we check for the ones whose field definition can't be modified.
+    // Some of those fields should be available though, so we let them through.
+    if (!field.edit_flag && !allowedFields.includes(field.key)) continue;
+
+    // Most subfields can not be edited, but some can be
+    if (field.is_subfield && !field.key.match(subfieldsRegex)) continue;
 
-const mapTypesToGrouparoo = (
+    const type = mapTypesToGrouparoo(field.field_type, field.name);
+    if (type) {
+      out.push({
+        key: field.name,
+        pipedriveKey: field.key,
+        required: required.includes(field.key),
+        important: field.important_flag, // they have a built-in (paid) feature to mark fields as important
+        groupMembershipField: field.name.startsWith(GROUP_FIELD_PREFIX), // used to hide our fields from destinationMappingOptions
+        type,
+      });
+    }
+  }
+
+  return out;
+};
+
+export const mapTypesToGrouparoo = (
   pipedriveType: string,
   fieldName: string
 ): DestinationMappingOptionsResponseType => {
   // https://pipedrive.readme.io/docs/core-api-concepts-custom-fields#section-types-of-custom-fields
 
   const overrides: Record<string, DestinationMappingOptionsResponseType> = {
     Email: "email",
+    Name: "string",
```

Comment:
What did this want to be otherwise?

Reply:
Ahaaa, I got it, I had a misconception about the usage of it.